decentraland/unity-renderer
# feat: PoC - never lock the mouse mode

## What does this PR change?
Closes #3877 

### Logic
- added `leftMouseButtonCursorLock` to `DataStore_Camera` to control new mode
- adjusted logic in `MouseCatcher.OnPointerDown` to be depended on this variable for cursor lock logic

### UI
- added `leftMouseButtonCursorLock `in `GeneralSettings`, so this setting get saved
- created `CameraMovementModeControlController` class and respective asset, 
- hocked up it in the `CameraMovementModeControlConfiguration.asset` (DropDown). Set text for dropdown there
- hocked up it to the `InputWidget.asset`, so it is showed in the settings near to "mouse inversion" and "mouse sensetivity" settings in UI.

- small clean-up and renamings
- commented file header in .editorconfig, to remove empty header being always created for new classes

## How to test the changes?
NOTE: not testable on Firefox!

1. Go to https://play.decentraland.zone/?renderer-branch=feat/never-lock-mouse-mode
2. Open ExploreV2 Menu -> Settings.
3. Change  "Camera movement mode" to "Hold right mouse button"
4. Now you should not be able to lock cursor by Left-Mouse-Button (LMB) click
5. Camera control is only possible when you hold Right-Mouse-Button (RMB). Cursor is also locked during RMB hold

6. Switch setting back and observe normal cursor-lock behavior with LMB 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
